gradio/image_utils.py

```diff
@@ -48,18 +48,18 @@ def format_image(
 
 
 def save_image(
-    y: np.ndarray | PIL.Image.Image | str | Path, cache_dir: str, format: str = "png"
+    y: np.ndarray | PIL.Image.Image | str | Path, cache_dir: str, format: str = "webp"
 ):
-    # numpy gets saved to png as default format
-    # PIL gets saved to its original format if possible
     if isinstance(y, np.ndarray):
         path = processing_utils.save_img_array_to_cache(
             y, cache_dir=cache_dir, format=format
         )
     elif isinstance(y, PIL.Image.Image):
+        fmt = y.format if y.format is not None and y.format != "png" else "webp"
```

Comment:
I specifically wanted to coerce png to webp because of the performance improvements. If we don't have some kind of handling here then it is will use the existing file format which will always be `png` for images generated in the browser.